ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -197,7 +211,11 @@ async def __call__(self, scope, receive, send):
                     break
             method = scope.get("method")
             server = scope.get("server")
-            scheme = scope.get("scheme", "http")
+            # Ensure HTTP requests always have the correct scheme
+            if scope["type"] == "http":
+                scheme = "http"
+            else:
+                scheme = scope.get("scheme", "http")
```

Comment:
Scheme and type capture different values. scheme is likely "http", "https", websocket, etc. and type can be http, websocket, http.disconnect, http.request,  some_custom_event1, etc. Using `scope["type"]` to set scheme feels a bit off. Docs: https://asgi.readthedocs.io/en/latest/specs/www.html.  Different web frameworks can define their own types. We should add simple websocket tests for all integrations that are using the agsi middleware to make sure it produces the expected spans.